src/config.rs

```diff
@@ -82,6 +82,50 @@ macro_rules! create_config {
             $(pub $i: Option<$ty>),+
         }
 
+        // This trait and the following impl blocks are there only so that we
+        // can use UCFS inside the get_docs() function on builtin types for configs.
+        trait IsConfigType {
+            fn get_variant_names() -> Vec<&'static str>;
```

Comment:
It's kind of a pity this only works for our enums. 

Maybe it'd be nicer to have a trait which simply provides a single `&'static str` which we could implement for all config types. We don't use the individual variant names anyway.